public/admin.php

```diff
@@ -98,6 +99,44 @@
     return back()->withErrors(__('legacy.error.error'));
 }
 
+if ($action === 'copy-unlocks') {
+    $fromAchievementIds = explode(',', requestInputSanitized('s'));
+    $fromAchievementCount = count($fromAchievementIds);
+    $toAchievementIds = explode(',', requestInputSanitized('a'));
+
+    // determine which players have earned all of the required achievements
+    $existing = PlayerAchievement::whereIn('achievement_id', $fromAchievementIds)
+        ->select(['user_id',
+            DB::raw('count(unlocked_at) AS softcore_count'),
+            DB::raw('count(unlocked_hardcore_at) AS hardcore_count'),
+            DB::raw('max(unlocked_at) AS unlocked_softcore_at'),
+            DB::raw('max(unlocked_hardcore_at) AS unlocked_hardcore_at'),
+        ])
+        ->groupBy('user_id')
+        ->having('softcore_count', '=', $fromAchievementCount)
+        ->get();
+
+    // award the target achievements, copying the unlock times and hardcore state
+    $unlockerId = request()->user()->id;
+    foreach ($existing as $playerAchievement) {
+        $hardcore = ($playerAchievement->hardcore_count == $fromAchievementCount);
+        $timestamp = Carbon::parse($hardcore ? $playerAchievement->unlocked_hardcore_at : $playerAchievement->unlocked_softcore_at);
+        foreach ($toAchievementIds as $nextID) {
```

Comment:
minor: `$toAchievementId` instead of `$nextID`?

Reply:
That's copied from https://github.com/RetroAchievements/RAWeb/blob/5c28b94e1c6b0d99b9ff2a41bc8f8be5743f6518/public/admin.php#L76-L84, but I can change it.